crates/burn-import/src/onnx/dim_inference.rs

```diff
@@ -120,6 +122,47 @@ fn constant_update_outputs(node: &mut Node) {
     };
 }
 
+/// Infer the shape of the output of a node with an explicit shape attribute
+///
+/// This includes the `RandomUniform`, `RandomNormal` operators
+///
+/// Also reads & interprets an optional `dtype` attribute
+fn explicit_shape_update_output(node: &mut Node) {
+    let dtype = node
+        .attrs
+        .get("dtype")
+        .map(|val| DataType::from_i32(val.clone().into_i32()).unwrap())
+        .unwrap_or(DataType::FLOAT);
+
+    let mut shape = node
+        .attrs
+        .get("shape")
+        .expect("required shape attribute missing")
+        .clone()
+        .into_i64s();
+
+    let elem_type = match dtype {
+        DataType::FLOAT => ElementType::Float32,
+        DataType::INT32 => ElementType::Int32,
+        DataType::INT64 => ElementType::Int64,
+        DataType::DOUBLE => ElementType::Float64,
+        DataType::BOOL => ElementType::Bool,
+        _ => panic!("tensor with type {dtype:?} not supported"),
+    };
```

Comment:
Might want to restrict to the actual dtypes supported for random distributions (i.e. floating points).